### PR TITLE
docs(contributing): add instructions on using clippy

### PR DESCRIPTION
**Pull request by KevinWMatthews**
Tell the developer to run clippy so that their changes don't fail CI.

Signed-off-by: Kevin W Matthews <kevinwmatthews@gmail.com>

### CODE REVIEW

**Comment by KevinWMatthews** [comment]
Sorry these keep coming in piecewise - I keep thinking I'm done.

[circleci runs clippy](https://github.com/interledger-rs/interledger-rs/blob/9f717878be89e495ca4795213dc30671747818f6/.circleci/config.yml#L54) but the contributing docs don't warn developers of this.

This directs developers to run `cargo clippy --all-targets --all-features -- --deny warnings` based on the circleci command.

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -42,6 +42,9 @@ Please follow the instruction below when making pull requests.
 - Make sure to run `cargo fmt` before you commit (required to pass CI).
     - To install rustfmt, run `rustup component add rustfmt`
     - If you would like to make your local setup reject unformatted commits, you can add `cargo fmt -- --check` as a pre-commit hook in the file `interledger-rs/.git/hooks/pre-commit`.
+- Make sure to run `cargo clippy` before you commit (required to pass CI).
+    - To install clippy, run `rustup component add clippy`
+    - If you would like to make your local setup reject unformatted commits, you can add `cargo clippy --all-targets --all-features -- --deny warnings` as a pre-commit hook in the file `interledger-rs/.git/hooks/pre-commit`.
```
Comment (dora-gt) [inline docs/CONTRIBUTING.md]: I would say, first we should test it with `clippy`, and then mention how we integrate it into pre-commit.

Comment (KevinWMatthews) [reply docs/CONTRIBUTING.md]: Makes sense. Pushed a change - is this what you were hoping for? Feel free to make another suggestion/edit.

If you like this update, should we add a similar line for the `cargo fmt` instructions?

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -42,6 +42,10 @@ Please follow the instruction below when making pull requests.
 - Make sure to run `cargo fmt` before you commit (required to pass CI).
     - To install rustfmt, run `rustup component add rustfmt`
     - If you would like to make your local setup reject unformatted commits, you can add `cargo fmt -- --check` as a pre-commit hook in the file `interledger-rs/.git/hooks/pre-commit`.
+- Make sure to run `cargo clippy` before you commit (required to pass CI).
+    - To install clippy, run `rustup component add clippy`
```
Comment (dora-gt) [inline docs/CONTRIBUTING.md]:
```suggestion
    - To install clippy, run `rustup component add clippy`.
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -42,6 +42,10 @@ Please follow the instruction below when making pull requests.
 - Make sure to run `cargo fmt` before you commit (required to pass CI).
     - To install rustfmt, run `rustup component add rustfmt`
```
Comment (dora-gt) [inline docs/CONTRIBUTING.md]:
```suggestion
     - To install rustfmt, run `rustup component add rustfmt`.
     - To run rustfmt, use `cargo fmt -- --check `.
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -42,6 +42,10 @@ Please follow the instruction below when making pull requests.
...
     - If you would like to make your local setup reject unformatted commits, you can add `cargo fmt -- --check` as a pre-commit hook in the file `interledger-rs/.git/hooks/pre-commit`.
```
Comment (dora-gt) [inline docs/CONTRIBUTING.md]:
```suggestion
     - If you would like to make your local setup reject unformatted commits, you can add the command as a pre-commit hook in the file `interledger-rs/.git/hooks/pre-commit`.
```

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -42,6 +42,10 @@ Please follow the instruction below when making pull requests.
...
+- Make sure to run `cargo clippy` before you commit (required to pass CI).
+    - To install clippy, run `rustup component add clippy`
+    - To run clippy, use `cargo clippy --all-targets --all-features -- -D warnings`.
+    - If you would like to make your local setup reject unformatted commits, you can add `cargo clippy --all-targets --all-features -- --deny warnings` as a pre-commit hook in the file `interledger-rs/.git/hooks/pre-commit`.
```
Comment (dora-gt) [inline docs/CONTRIBUTING.md]:
```suggestion
    - If you would like to make your local setup reject unformatted commits, you can add the command as a pre-commit hook in the file `interledger-rs/.git/hooks/pre-commit`.
```

**Comment by KevinWMatthews** [comment]
Rebased onto the latest master.

**Pull request merged**